src/main/java/net/sf/jabref/logic/autosaveandbackup/BackupManager.java

```diff
@@ -98,6 +96,13 @@ private void shutdown() {
         }
     }
 
+    static Path getBackupPath(Path originalPath) {
+        Objects.requireNonNull(originalPath.getParent());
+        Path oldFileName = originalPath.getFileName();
+        String newFileName = oldFileName.toString() + BACKUP_FILENAME_ENDING;
```

Comment:
What about [Path.ChangeExtension](https://msdn.microsoft.com/en-us/library/system.io.path.changeextension(v=vs.110).aspx). Ahhh damn, we are in the Java world, so:
probably `return originalPath.resolveSibling(BackupEnding);` (or something similar, but there should be a way to change the extension without converting to string)
And we should probably extract it to a helper class (because its used not only for autosave but also backup).